packages/environment/test/connectOrStartTest.js

```diff
@@ -0,0 +1,54 @@
+const assert = require("chai").assert;
+const Develop = require("../develop");
+
+describe("connectOrStart test network", async function () {
+  const ipcOptions = { network: "test" };
+  const ganacheOptions = {
+    host: "127.0.0.1",
+    network_id: 666,
+    port: 6969
+  };
+
+  it("starts Ganache when no Ganache instance is running", async function () {
+    let connection;
+    try {
+      connection = await Develop.connectOrStart(ipcOptions, ganacheOptions);
+      assert.isTrue(connection.started, "A new server did not spin up");
+      assert.isFunction(connection.disconnect, "Disconnect is not a function");
+    } catch (error) {
+      console.log(error);
+      assert.fail("Develop.connectOrStart should not have thrown!");
+    } finally {
+      if (connection) {
+        return connection.disconnect();
+      }
+    }
+  });
+
+  it("connects to an established ganache instance", async function () {
+    let connectionOneDisconnect, connectionTwoDisconnect;
+
+    try {
+      //establish a connection
+      connectionOneDisconnect = (
+        await Develop.connectOrStart(ipcOptions, ganacheOptions)
+      ).disconnect;
+
+      //invoke the method again
+      const result = await Develop.connectOrStart(ipcOptions, ganacheOptions);
+      connectionTwoDisconnect = result.disconnect;
+      assert.isFalse(result.started);
+    } catch (error) {
+      console.log(error);
+      assert.fail("Develop.connectOrStart should not have thrown!");
```

Comment:
You can omit the `catch` here, i think... the reason the test fails should be apparent in the error logs reported by the test runner. It seems strange to me to catch the actual error and then re-throw something else; it could make debugging the reason it failed in the first place more difficult.

Reply:
Oh, yeah, I missed this, but I want to add my agreement here.  Catch-and-rethrow or catch-and-fail is generally a bad idea for exactly this reason, it obscures the error's origin.  If you want throwing to cause a failure, I would also say that the way to do that is by just not catching at all.

Reply:
To be clear if you need to do some stuff before rethrowing the *original* error that makes sense, it's throwing a *new* one, as with `assert.fail()` here, that's the problem.  In this case you log the original error before failing?  Which does avoid losing that information, but ultimately you're not accomplishing that couldn't be accomplished more simply by not catching in the first place.

Reply:
Thanks @haltman-at, @davidmurdoch. I agree that not catching is better in this case. Updated!